src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -188,19 +188,29 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, train_val_dmatrix, model_di
     :param is_master: True if single node training, or the current node is the master node in distributed training.
     """
     # Parse arguments for train() API
-    early_stopping_rounds = train_cfg.pop('early_stopping_rounds', None)
     num_round = train_cfg.pop("num_round")
+    # Parse arguments for intermediate model callback
+    save_model_on_termination = train_cfg.pop('save_model_on_termination', "false")
 
     # Evaluation metrics to use with train() API
     tuning_objective_metric_param = train_cfg.pop("_tuning_objective_metric", None)
     eval_metric = train_cfg.get("eval_metric")
-    cleaned_eval_metric, configured_feval = train_utils.get_eval_metrics_and_feval(
+    cleaned_eval_metric, configured_feval, tuning_objective_metric = train_utils.get_eval_metrics_and_feval(
         tuning_objective_metric_param, eval_metric)
     if cleaned_eval_metric:
         train_cfg['eval_metric'] = cleaned_eval_metric
     else:
         train_cfg.pop('eval_metric', None)
 
+    early_stopping_rounds = train_cfg.pop('early_stopping_rounds', None)
+    early_stopping_data = 'validation' if val_dmatrix else 'train'
+    early_stopping_metric = None
+    if early_stopping_rounds:
+        if tuning_objective_metric:
+            early_stopping_metric = tuning_objective_metric[-1]
```

Comment:
And what's the reasoning for choosing the last metric to use for early stopping?

Reply:
Both eval_metric and tuning_objective_metric have been there for long time. I agree that they are kind of misleading. But it is XGB's naming convention. I prefer to keep it as.

It is also the convention to choose the last metric as early stopping metric before. So we keep it as is so that we don't break customer's code.